### PR TITLE
prov/tcp: Bind a specific port range, using SO_REUSEADDR

### PR DESCRIPTION
**Pull request by nikitaxgusev**
**Issue:**
While we try to bind to a port after several unexpected interrupts, while the app is in process, the port stays in `TIME_WAIT` status. It is the specific feature of Linux OS, `TIME_WAIT` state is what a socket goes into when it closes. We need to wait, when OS let the specific port  `LISTEN` again because the app, you try to launch just failed and cant be launched.
**Fixed:**
That is why we should use `SO_REUSEADDR`(calling `tcpx_setup_socket()`). This socket option tells the kernel that even if this port is busy (in the TIME_WAIT state), go ahead and reuse it anyway.  If it is busy,  but with another state, you will still get an address already in use  error.
**Tested:**
Since [ae62148](https://github.com/ofiwg/libfabric/commit/ae62148c8d73b8bdd001f15c87e97ca7583720c9) we haven't use `SO_REUSEADDR`. We missed `TIME_WAIT` problem and let `tcpx_setup_socket_nodelay()` be without `SO_REUSEADDR` and use only `TCP_NODELAY`.
We've tested now on different port ranges and try to make unexpected interrupts (Ctr+c) and the app successfully reuse the specific port even it has `TIME_WAIT`.

Signed-off-by: Nikita Gusev <nikita.gusev@intel.com>

### CODE REVIEW

File: prov/tcp/src/tcpx_ep.c [hunk 1]
```diff
@@ -253,20 +239,14 @@ static int tcpx_pep_sock_create(struct tcpx_pep *pep)
...
-			goto err;
-		}
 		ret = bind(pep->sock, pep->info->src_addr,
-		      (socklen_t) pep->info->src_addrlen);
+			  (socklen_t) pep->info->src_addrlen);
 	} else {
-		ret = tcpx_setup_socket_nodelay(pep->sock);
-		if (ret) {
-			goto err;
-		}
```
Comment (shefty) [inline prov/tcp/src/tcpx_ep.c]: This is reverting a change made by this commit:
https://github.com/ofiwg/libfabric/commit/ae62148c8d73b8bdd001f15c87e97ca7583720c9
tcpx_bind_to_port_range() searches for an available port to use.  Why do we want to set that port to re-usable?

**Comment by nikitaxgusev** [comment]
@shefty Sorry, Sean, for unreadable commit message. Now I am more experienced and will try to explain well next times. :) Please, check it again, I tried to explain the issue. Thanks.

**Comment by shefty** [comment]
You added a comment to the  github pull request.  However, the actual commit message is still empty.  That is what needs to be updated, as it will be part of the git log.  Please copy the text from github into the commit message

**Comment by jsquyres** [comment]
@nikitaxgusev Specifically, you can probably fix your commit with `git commit --amend`, and then you can force push it up to your github fork (updating a pull request is just about the *only* socially-acceptable reason to git force push!).

**Comment by nikitaxgusev** [comment]
@shefty The commit message has been added.

**Comment by nikitaxgusev** [comment]
@jsquyres Thanks, done.

**Comment by vkrishna** [comment]
Looks good to me. 

I believe the motivation for this change is to get same port in the port range if the app restarted.

**Pull request merged**